internal/cache/db_test.go

```diff
@@ -85,21 +87,24 @@ func TestNew(t *testing.T) {
 				require.NoError(t, err, "Setup: Can't update the file with invalid db content")
 			}
 
-			expiration, err := time.Parse(time.DateOnly, "2003-01-01")
+			expiration, err := time.Parse(time.DateOnly, "2004-01-01")
 			require.NoError(t, err, "Setup: could not calculate expiration date for tests")
 
 			cacheOpts := []cache.Option{cache.WithExpirationDate(expiration)}
 			if tc.expirationDate != "" {
 				expiration, err = time.Parse(time.DateOnly, tc.expirationDate)
 				require.NoError(t, err, "Setup: could not calculate expiration date for tests")
-				cacheOpts = []cache.Option{cache.WithExpirationDate(expiration)}
+				cacheOpts = append(cacheOpts, cache.WithExpirationDate(expiration))
```

Comment:
This is not what I meant on my previous comment.
Set a default expiration, then, if tc.expirationDate != "", compute the new expiration,. and after closing the if, do:
`cacheOpts := []cache.Option{cache.WithExpirationDate(expiration)}`

Reply:
Oops, sorry... I really misunderstood this one completely.